src/HotChocolate/ElasticSearch/src/Data.ElasticSearch.Driver/Filters/Handlers/ElasticSearchInOperationHandler.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System.Collections;
+using HotChocolate.Configuration;
+using HotChocolate.Data.Filters;
+using HotChocolate.Language;
+using HotChocolate.Types;
+
+namespace HotChocolate.Data.ElasticSearch.Filters;
+
+/// <summary>
+/// This filter operation handler maps a In operation field to a
+/// <see cref="ISearchOperation"/>
+/// </summary>
+public class ElasticSearchInOperationHandler
+    : ElasticSearchOperationHandlerBase
+{
+    /// <summary>
+    /// Initializes a new instance of <see cref="ElasticSearchInOperationHandler"/>
+    /// </summary>
+    public ElasticSearchInOperationHandler(InputParser inputParser)
+        : base(inputParser)
+    {
+    }
+
+    /// <inheritdoc />
+    public override bool CanHandle(
+        ITypeCompletionContext context,
+        IFilterInputTypeDefinition typeDefinition,
+        IFilterFieldDefinition fieldDefinition)
+        => fieldDefinition is FilterOperationFieldDefinition { Id: DefaultFilterOperations.In };
+
+    /// <inheritdoc />
+    public override ISearchOperation HandleOperation(
+        ElasticSearchFilterVisitorContext context,
+        IFilterOperationField field,
+        IValueNode value,
+        object? parsedValue)
+    {
+
+        if (parsedValue is null)
+        {
+            throw new InvalidOperationException();
+        }
+
+        var enumerable = ((IEnumerable)parsedValue).Cast<object>().ToList();
+        var shouldOperations = enumerable
+            .Select(val => new MatchOperation(context.GetPath(), ElasticSearchOperationKind.Filter, val.ToString()))
+            .ToList();
```

Comment:
I remove ToList call and replace the properties with enumerables. 

The third parameter for MatchOperation is of type string. Therefore I call ToString. I think in the end ES works on strings anyway. I guess it will work.

Reply:
I think we missunderstood each other here

```
   var enumerable = ((IEnumerable)parsedValue).Cast<object>().ToList(); <---- I meant this to list :) 
        var shouldOperations = enumerable
            .Select(val => new MatchOperation(context.GetPath(), ElasticSearchOperationKind.Filter, val.ToString()))
            .ToList();
```

Reply:
I will change it back and adjust it the way you suggested it.

Reply:
I changed it back